pipelines/library/src/common/VersionInfo.groovy

```diff
@@ -148,61 +150,76 @@ class VersionInfo {
 
     // Match JDK9+
     private boolean match223(versionString) {
+        // jdk9+ version string format:
+        //   jdk-<major>[.<minor>][.<security>][.<patch>][-<pre>]+<build>[-<opt>]
+        // eg.
+        // jdk-15+36
+        // jdk-11.0.9+11
+        // jdk-11.0.9.1+1
+        // jdk-11.0.9-ea+11
+        // jdk-11.0.9+11-202011050024
+        // jdk-11.0.9+11-adhoc.username-myfolder
+        //
         //Regexes based on those in http://openjdk.java.net/jeps/223
-        // Technically the standard supports an arbitrary number of numbers, we will support 3 for now
-        final vnumRegex = "(?<major>[0-9]+)(\\.(?<minor>[0-9]+))?(\\.(?<security>[0-9]+))?"
-        final preRegex = "(?<pre>[a-zA-Z0-9]+)"
+        // Note, currently openjdk only uses the first regex format of 223
+        // Technically the standard supports an arbitrary number of numbers, we will support 4 for now
+        final vnumRegex = "(?<major>[0-9]+)(\\.(?<minor>[0-9]+))?(\\.(?<security>[0-9]+))?(\\.(?<patch>[0-9]+))?"
+        final preRegex = "(?<pre>[a-zA-Z]+)"
         final buildRegex = "(?<build>[0-9]+)"
         final optRegex = "(?<opt>[-a-zA-Z0-9\\.]+)"
 
-        List<String> version223Regexs = [
-                "(?:jdk\\-)?(?<version>${vnumRegex}(\\-${preRegex})?\\+${buildRegex}(\\-${optRegex})?)".toString(),
-                "(?:jdk\\-)?(?<version>${vnumRegex}\\-${preRegex}(\\-${optRegex})?)".toString(),
-                "(?:jdk\\-)?(?<version>${vnumRegex}(\\+\\-${optRegex})?)".toString()
-        ]
-
-        for (String regex : version223Regexs) {
-            context.println "[INFO] Attempting to match223 regex: ${regex}"
-            final matched223 = versionString =~ /^${regex}.*/
-            if (matched223.matches()) {
-                context.println "[SUCCESS] match223 regex matched!"
-
-                major = or0(matched223, 'major')
-                minor = or0(matched223, 'minor')
-                security = or0(matched223, 'security')
-
-                if (matched223.group('pre') != null) {
-                    pre = matched223.group('pre')
-                    context.println "[SUCCESS] regex group (pre) of last 223 regex matched to ${pre}"
-                }
+        String regex = "(?:jdk\\-)?(?<version>${vnumRegex}(\\-${preRegex})?\\+${buildRegex}(\\-${optRegex})?)".toString()
```

Comment:
I think this may make build tags required, where as under the previous regex, the second one did not require a build